lib/core/utils/preload-cssom.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Returns a then(able) queue of CSSStyleSheet(s)
+ * @param {Object} ownerDocument document object to be inspected for stylesheets
+ * @param {number} timeout on network request for stylesheet that need to be externally fetched
+ * @param {Function} getSheetFromTextFn a utility function to generate a style sheet from text
+ * @return {Object} queue
+ * @private
+ */
+function loadCssom(ownerDocument, timeout, getSheetFromTextFn) {
+	const q = axe.utils.queue();
+
+	Array.from(ownerDocument.styleSheets).forEach(sheet => {
+		if (sheet.disabled) {
+			return;
+		}
+
+		try {
+			// attempt to resolve if sheet is relative/ same domain
+			sheet.cssRules && q.defer(resolve => resolve(sheet));
+		} catch (e) {
+			const deferredSheet = (resolve, reject) => {
```

Comment:
Why do we need the xhr call at all? Won't creating the `<style>` element in the head of the new document cause the file to be loaded? Won't the file already be in the cache because the page loaded? You will need to explain the logic of how this all works to me.